docs/concepts/time-and-windows-in-ksql-queries.rst

```diff
@@ -0,0 +1,305 @@
+    to wait before closing a window. Retention time is still a valid parameter
+    that defines for how long the potentially closed window is stored. This
+    enables accessing the window with interactive queries, even if it's closed.
+
+In the real world, late-arriving records are always possible, and your KSQL
+applications must account for them properly. The system's time semantics
+determine how late records are handled. For processing-time, the semantics are
+“when the record is being processed”, which means that the notion of late records
+isn't applicable because, by definition, no record can be late.
+
+Late-arriving records are considered “late” only for event-time and ingestion-time
+semantics. In both cases, KSQL is able to handle late-arriving records properly.
+
+.. important::
+
+    KSQL is based on the Unix epoch time in the UTC timezone, and this can affect
+    time windows. For example, if you define a 24-hour tumbling time window, it will
+    be in the UTC timezone, which may not be appropriate if you want to have daily
+    windows in your timezone.
+
+Window Types
+============
+
+There are three ways to define time windows in KSQL.
+
+.. image:: ../img/ksql-window-aggregation.png
+   :alt: Diagram showing three types of time windows in KSQL streams: tumbling, hopping, and session
+
+Tumbling Window
+---------------
+
+All time windows are the same size and adjacent to each other, which
+means that whenever a window ends, the next window starts.
+
+For example, if you want to compute the the five highest-value orders
+per zip code per hour in an ``orders`` stream, you might run a query like this:
+
+.. code:: sql
+
+    SELECT orderzip_code, TOPK(order_total, 5) FROM orders
+      WINDOW TUMBLING (SIZE 1 HOUR) GROUP BY order_zipcode;
+
+Here's another example: to detect potential credit card fraud in an
+``authorization_attempts`` stream, you might run a query for the number of
+authorization attempts on a particular card that's greater than three, during
+a time interval of five seconds.
+
+.. code:: sql
+
+    SELECT card_number, count(*) FROM authorization_attempts
+      WINDOW TUMBLING (SIZE 5 SECONDS)
+      GROUP BY card_number HAVING count(*) > 3;
+
+The tumbling window's start time is inclusive, but the end time is exclusive.
+This is important for non-overlapping windows, in which each record must be
+contained in exactly one window.
+
+Hopping Window
+--------------
+
+All time windows are the same size, but they might overlap, depending on the
+length of time specified in the ADVANCE BY property.
+
+For example, if you want to count the pageviews for only ``Region_6`` by female
+users for a hopping window of 30 seconds that advances by 10 seconds, you might
+run a query like this:
+
+.. code:: sql
+
+    SELECT regionid, count(*) FROM pageviews 
+      WINDOW HOPPING (SIZE 30 SECONDS, ADVANCE BY 10 SECONDS)
+      WHERE UCASE(gender)='FEMALE' AND LCASE (regionid) LIKE '%_6'

```

Comment:
`LIKE '%_6'` is a broader predicate than `Region_6`, as it can also match `Foo_6`, for example.  Calling this out because your textual description explicitly says `Region_6`.

So the SQL should be:
```
WHERE UCASE(gender)='FEMALE' AND regionid = 'Region_6'
```

Or update the description ;-)